### ISSUE TITLE
Treat detached panel windows as first-class (fix IntGrid paint lock)

### ISSUE THREAD

**Issue by duxtinto** [deepnight/led#1232]
## Summary

Detached browser windows (Map, Tilesets, Layers) are still implemented as modals with flag hacks. That causes **canvas painting to stop** when a separate window is open (notably IntGrid painting with Map/tileset windows detached), because those panels survive `closeAll` via `canBeClosedManually = false` while still `countAsModal() = true`, which makes `App.isLocked()` true.

**Goal:** treat external windows as a first-class hosting mode, not a special class of blocking modals.

## Why painting broke

```mermaid
flowchart LR
  detach["Panel.detach()"] --> flags["canBeClosedManually = false"]
  flags --> survive["Survives Modal.closeAll"]
  flags --> lock["hasAnyUnclosable = true"]
  lock --> locked["App.isLocked()"]
  locked --> block["Editor.onMouseDown returns early"]
```

- `MapPanel` opts out with `countAsModal() = false`, so Map alone does not lock.
- `EditTilesetDefs` / `EditLayerDefs` keep default `countAsModal() = true` while detached with `canBeClosedManually = false`.
- `App.isLocked()` uses `Modal.hasAnyUnclosable()` → canvas painting stops in `Editor.onMouseDown`.
- Keeping tileset detail windows open across tool switches made details windows unclosable-by-`closeAll`, which also locks the editor.

Docked side panels should still behave as exclusive modals. Detached OS windows should not.

## Approach

Introduce **`ui.modal.DetachablePanel extends Panel`** as the single place for detach hosting. Subclasses stay panels when docked; when detached they become companion windows.

### 1. New base: `DetachablePanel`

Add `src/electron.renderer/ui/modal/DetachablePanel.hx` owning the duplicated machinery from Map/Tilesets/Layers:

- `window.open`, style clone, reparent `jModalAndMask`, dock, `beforeunload`, detached resize (`100vw`/`100vh`)
- detach button chrome hooks
- `isDetached()`, `focusDetachedWindow()`, abstract/overridable window name + title

**Semantics when detached:**

| Concern | Docked | Detached |
|--------|--------|----------|
| `countAsModal()` | subclass default (Map stays `false`) | **always `false`** |
| Survive `closeAll` | normal | via `countAsModal=false` (no `canBeClosedManually` hack) |
| `App.isLocked` / shortcuts | as today | never blocked by this panel |
| Window close | n/a | details: dismiss; full panels: dock (existing behavior) |

Override `countAsModal()` in the base as:

```haxe
override function countAsModal() {
  return !isDetached() && super.countAsModal();
}
```

Map keeps its own override returning `false` always (docked Map is already a non-blocking companion).

Drop temporary `canBeClosedManually` flip-flops where they only existed to survive `closeAll` while opening another panel (`openDetailsInSeparateWindow`, `openLayerDefInPanel`), once detached/non-blocking panels are correctly excluded from `closeAll` via `countAsModal`.

### 2. Migrate the three panels

- `MapPanel extends DetachablePanel` — keep Map-specific layout (`setMapPanelDetached`), embedded rules, palette pop-in-before-detach.
- `EditTilesetDefs extends DetachablePanel` — keep `detailsOnly`, multi-details windows, list vs details window names.
- `EditLayerDefs extends DetachablePanel` — keep fullscreen + detach.

Delete the triplicated `detachPanel` / `finishDetachPanel` / `dockPanel` / `applyDetachedStyles` / `postUpdate` stylesheet wait loops from each file.

### 3. Editor input: treat any detached companion as first-class

Generalize Map-only focus/FPS hacks in `App.hx`:

- `DetachablePanel.hasAnyDetached()` (or registry) instead of only `MapPanel.isDetachedMap()` for:
  - uncapped FPS while main window blurred
  - canvas `mouseenter`/`mousedown` → `requestCanvasFocus()`
- IntGrid value pick in `IntGridPalette.hx`: after `selectValue`, call `requestCanvasFocus()` when the palette lives in a detached document (same pattern as `Tileset.hx`).

### 4. Keep `closeAll` behavior for docked tools

Docked Project / Layers / Tilesets / Map still close each other when switching toolbar tools via `Panel`’s constructor `closeAll`. Detached windows remain open because they no longer count as closable modals.

## Implementation todos

- [ ] Add `DetachablePanel` with shared detach/dock and `countAsModal=!isDetached`
- [ ] Migrate `MapPanel`, `EditTilesetDefs`, `EditLayerDefs` onto `DetachablePanel`
- [ ] Generalize App FPS/canvas focus for any detached companion; IntGrid `requestCanvasFocus`
- [ ] Remove `canBeClosedManually` protect/restore flip-flops that only existed for `closeAll` survival
- [ ] Manually verify IntGrid paint + tileset details surviving Map switch

## Out of scope

- Detaching Project / Entities / Enums / Help
- Rewriting `Dialog` / `ContextMenu` hosting (existing `setHostFromElement` stays)
- Changing Electron `window.open` for a BrowserWindow API

## Verification

1. Detach Map, select IntGrid value, paint on canvas — works immediately.
2. Open tileset details window(s), switch to Map toolbar — details stay open **and** painting still works.
3. Detach Layers / full Tilesets list — editor remains unlocked; Escape still closes docked panels, not detached windows.
4. Close a tileset details OS window — that panel dismisses; docked Map (if open) stays.
5. Dock Map back — layout / canvas focus behave as before.

## Key files

- `src/electron.renderer/ui/Modal.hx`
- `src/electron.renderer/ui/modal/Panel.hx`
- `src/electron.renderer/ui/modal/panel/MapPanel.hx`
- `src/electron.renderer/ui/modal/panel/EditTilesetDefs.hx`
- `src/electron.renderer/ui/modal/panel/EditLayerDefs.hx`
- `src/electron.renderer/App.hx`
- `src/electron.renderer/page/Editor.hx`
- `src/electron.renderer/ui/palette/IntGridPalette.hx`

**Comment by duxtinto** [comment]
sorry, this was a mistake
